.github/workflows/build-and-test.yml

```diff
@@ -156,15 +156,14 @@ jobs:
       - name: Run the tests
         run: |
           date
-          # Cache LXD images.
-          # First, for containers.
-          lxc launch ubuntu:22.04 temp
-          lxc stop temp
-          lxc delete temp
-          # Secondly, for VM's.
-          lxc launch ubuntu:22.04 temp --vm -c limits.cpu=2 -c limits.memory=4GiB -d root,size=25GiB
-          lxc stop temp
-          lxc delete temp
+          # Cache LXD images for both VM and containers.
+          lxc launch ubuntu:22.04 ctemp
+          lxc launch ubuntu:22.04 vmtemp --vm -c limits.cpu=2 -c limits.memory=4GiB -d root,size=25GiB
+          lxc stop ctemp
+          lxc delete ctemp
+          sleep 60
+          lxc stop vmtemp
+          lxc delete vmtemp
```

Comment:
Drive-by comments:
- wonder if this would make sense in main as well? 
-  to reduce blowing up the main workflow file too much I feel this should be added as a function to tests/scripts/actionutils.sh